pavelib/paver_tests/test_paver_bok_choy_cmds.py

```diff
@@ -167,3 +169,72 @@ def test_invalid_verbosity_and_processes(self):
         suite = BokChoyTestSuite('', num_processes=2, verbosity=3)
         with self.assertRaises(BuildFailure):
             BokChoyTestSuite.verbosity_processes_string(suite)
+
+
+class TestPaverPa11ycrawlerCmd(unittest.TestCase):
+
+    """
+    Paver pa11ycrawler command test cases.  Most of the functionality is
+    inherited from BokChoyTestSuite, so those tests aren't duplicated.
+    """
+
+    def setUp(self):
+        super(TestPaverPa11ycrawlerCmd, self).setUp()
+
+        # Mock shell commands
+        mock_sh = patch('pavelib.utils.test.suites.bokchoy_suite.sh')
+        self._mock_sh = mock_sh.start()
+
+        # Cleanup mocks
+        self.addCleanup(mock_sh.stop)
+
+    def _expected_command(self, report_dir):
+        """
+        Returns the expected command to run pa11ycrawler.
+        """
+        cms_start_url = (
+            "http://localhost:8031/auto_auth?redirect=true&course_id=course-v1"
```

Comment:
That's reasonable. Really, wherever you think is the most intuitive place...

On Wed, Apr 13, 2016 at 9:21 AM, Christine Lytwynec <
notifications@github.com> wrote:

> In pavelib/paver_tests/test_paver_bok_choy_cmds.py
> <https://github.com/edx/edx-platform/pull/12060#discussion_r59545790>:
>
> > +    def setUp(self):
> > +        super(TestPaverPa11ycrawlerCmd, self).setUp()
> > +
> > +        # Mock shell commands
> > +        mock_sh = patch('pavelib.utils.test.suites.bokchoy_suite.sh')
> > +        self._mock_sh = mock_sh.start()
> > +
> > +        # Cleanup mocks
> > +        self.addCleanup(mock_sh.stop)
> > +
> > +    def _expected_command(self, report_dir):
> > +        """
> > +        Returns the expected command to run pa11ycrawler.
> > +        """
> > +        cms_start_url = (
> > +            "http://localhost:8031/auto_auth?redirect=true&course_id=course-v1"
>
> Yea, that makes sense. It might make sense to store a list of start urls
> in envs.py?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-platform/pull/12060/files/a14718bb923283e43f1d5fba08546d2585e00261#r59545790>
>



-- 
*Ben Patterson*

edX | Engineering Manager  | ben@edx.org

141 Portland Street, 9th floor

Cambridge, MA 02139
http://www.edx.org <http://www.edxonline.org/>

[image:
http://www.e-learn.nl/media/blogs/e-learn/edX_Logo_Col_RGB_FINAL.jpg?mtime=1336074566]